File: haskell/bob/Bob.hs
module Bob(responseFor) where

import Data.Text(stripEnd, pack, unpack)
import Data.Char(isLower, isUpper)

responseFor :: String -> String
responseFor question = respond (unpack (stripEnd (pack question)))
    where
        respond question
            | question == "" = "Fine. Be that way!"
            | any isUpper question && not (any isLower question) = "Whoa, chill out!"
            | last question == '?' = "Sure."
            | otherwise = "Whatever."

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
    input = input.trim();
    if (input === '')
        return 'Fine. Be that way!';
    if (input === input.toUpperCase() && input.search(/[a-zA-Z]/) !== -1)
        return 'Whoa, chill out!';
    if (input.slice(-1) === '?')
        return 'Sure.';
    return 'Whatever.';
};

module.exports = Bob;
